### PR TITLE
Mark action results for purity clean

### PR DESCRIPTION
**Pull request by Hamishpk**
After further investigation errors seen from bucket.Exists() seem to occur when the blob doesn't exist. I have changed these to a log line so that purity only marks action results as live if bucket.Delete fails.

I've also added markActionResults as it's needed for purity clean.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Version 10.2.0
+-------------
     * Add timeouts to Elan's remote client
     * Add metrics to Elan's remote client
+	* Only return delete errors in Elan's gc
+	* Fix purity clean
```
Comment (fische) [inline ChangeLog:6]: I think you have to make a new version for the release pipeline to be triggered.

Comment (Hamishpk) [reply ChangeLog:6]: It looks like the release pipeline didn't produce any assets anyway so I was going to delete it before landing this

Comment (fische) [reply ChangeLog]: But still, are you sure it's gonna get triggered when merging this? I thought it was getting triggered if the `VERSION` is changed

Comment (Garbett1) [reply ChangeLog]: Looks like the release _attempted_ to trigger, but failed: https://github.com/thought-machine/please-servers/actions/runs/3311956028/jobs/5468074324#step:6:30

Comment (Hamishpk) [reply ChangeLog]: The release process is pretty flaky. I'm sure I've seen that error before!

I have bumped the version in this PR

**Review by fische — APPROVED** [review]

**Pull request merged**